pkg/image/api/helper.go

```diff
@@ -33,8 +33,9 @@ const (
 	containerImageEntrypointAnnotationFormatKey = "openshift.io/container.%s.image.entrypoint"
 )
 
-// TODO remove (base, tag, id)
-func parseRepositoryTag(repos string) (string, string, string) {
+// ParseRepositoryTag splits a string into its name component and either tag or id if present.
+// TODO remove
+func parseRepositoryTag(repos string) (base string, tag string, id string) {
```

Comment:
This is something different. Mentioned functions expect names of instances of types they are designed for. They don't expect fully qualified docker image reference.

This indeed deserves a better refactor. But I'd rather avoid doing it with this PR which is already big enough.